internal/testrunner/runners/pipeline/runner.go

```diff
@@ -103,6 +103,8 @@ func (r *runner) run() ([]testrunner.TestResult, error) {
 		}
 		startTime := time.Now()
 
+		// TODO: Add tests to cover regressive use of json.Unmarshal in loadTestCaseFile.
+		// See https://github.com/elastic/elastic-package/pull/717.
```

Comment:
Nit. These TODOs won't be needed if we add the sample package for the scripted tests. It is true that it'd be nice to increase unit testing coverage here, but maybe further refactors are needed for this.

Reply:
This is addressed in my comment above. I think that the test as it stands justifies leaving these here in case someone can come up with a simple but effective test that will capture regressions.